docs/book/beyond_basics/sample_webhook.md

```diff
@@ -0,0 +1,228 @@
+
+When the Server starts, it will register all Webhook Configurations with the apiserver to start accepting and
+routing requests to the Handlers.
+
+[controller-runtime](https://godoc.org/sigs.k8s.io/controller-runtime/pkg) provides a useful package for
+building a webhook.
+You can incrementally set the configuration of a webhook and then invoke `Build` to complete building a webhook.
+
+If you want to specify the name and(or) path for your webhook instead of using the default, you can invoke
+`Name("yourname")` and `Path("/yourpath")` respectively.
+
+{% sample lang="go" %}
+```go
+wh, err := builder.NewWebhookBuilder().
+    Mutating().
+    Operations(admissionregistrationv1beta1.Create).
+    ForType(&corev1.Pod{}).
+    Handlers(&podAnnotator{}).
+    WithManager(mgr).
+    Build()
+if err != nil {
+    // handle error
+}
+```
+{% endmethod %}
+
+
+#### Creating a Server
+
+{% method %}
+
+A Server registers Webhook Configuration with the apiserver and creates an HTTP server to route requests to the handlers.
+
+The server is behind a Kubernetes Service and provides a certificate to the apiserver when serving requests.
+
+The Server depends on a Kubernetes Secret containing this certificate to be mounted under `CertDir`.
+
+The Secret doesn't have to be pre-populated, the server can populate the secret with certificate.
+
+A new webhook server can be created by invoking `webhook.NewServer`.
+The Server will be registered to the provided manager.
+You can specify `Port`, `CertDir` and various `BootstrapOptions`.
+
+It is strongly recommended to set the `Secret` and `Service` fields.
+In other words, it is recommended to use a k8s secret to store the certificates and
+a k8s service to front this webhook server.
+
+{% sample lang="go" %}
+```go
+svr, err := webhook.NewServer("foo-admission-server", mgr, webhook.ServerOptions{
+    CertDir: "/tmp/cert",
+    BootstrapOptions: &webhook.BootstrapOptions{
+        Secret: &types.NamespacedName{
+            Namespace: "default",
+            Name:      "foo-admission-server-secret",
+        },
+
+        Service: &webhook.Service{
+            Namespace: "default",
+            Name:      "foo-admission-server-service",
+            // Selectors should select the pods that runs this webhook server.
+            Selectors: map[string]string{
+                "app": "foo-admission-server",
+            },
+        },
+    },
+})
+if err != nil {
+    // handle error
+}
+```
+{% endmethod %}
+
+#### Registering a Webhook with the Server
+
+You can register webhook(s) in the webhook server by invoking `svr.Register(wh)`.
+
+
+## Implementing Webhook Handler
+
+
+#### Implementing the Handler Business Logic
+
+{% method %}
+
+`decoder types.Decoder` is a decoder that knows how the decode all core type and your CRD types.
+
+`client client.Client` is a client that knows how to talk to the API server.
+
+Note: If using a caching client, you don't need explicitly tell the cache what resources to watch.
```

Comment:
Note: the client.Client provided to the InjectClient function reads from a cache which is lazily initialized.  To eagerly initialize the cache, perform a read operation with the client before starting the server.